test/benchmark_http_client_test.cc

```diff
@@ -310,8 +359,46 @@ TEST_F(BenchmarkClientHttpTest, BadContentLength) {
   };
 
   EXPECT_CALL(stream_encoder_, encodeData(_, _)).Times(0);
-  testBasicFunctionality(1, 1, 1);
+  auto client_setup_parameters = ClientSetupParameters(1, 1, 1, request_generator);
+  TestBenchmarkClientProcessesExpectedInflightRequests(client_setup_parameters);
   EXPECT_EQ(1, getCounter("http_2xx"));
 }
 
+TEST_F(BenchmarkClientHttpTest, RequestGeneratorProvidingDifferentPathsSendsRequestsOnThosePaths) {
+  std::vector<HeaderMapPtr> requests_for_generator_to_send;
+  const std::initializer_list<std::pair<std::string, std::string>> header_map_for_first_request{
```

Comment:
If I remember correctly, we discussed that we will try to avoid using the `std::initializer_list` explicitly. What changed our plans?

Reply:
I think the problem is that the constructor specifically takes an initializer list, and it makes sense to declare it as an initializer list for that. I think I can actually use auto here, but I'm not sure that it would be an improvement.